app/src/main/java/org/mozilla/fenix/gleanplumb/MessageNotificationWorker.kt

```diff
@@ -0,0 +1,125 @@
+ * Background [Worker] that polls Nimbus for available [Message]s at a given interval.
+ * A [Notification] will be created using the configuration of the next highest priority [Message]
+ * if it has not already been displayed.
+ */
+class MessageNotificationWorker(
+    context: Context,
+    workerParameters: WorkerParameters,
+) : Worker(context, workerParameters) {
+
+    @OptIn(DelicateCoroutinesApi::class) // GlobalScope usage
+    override fun doWork(): Result {
+        GlobalScope.launch(Dispatchers.IO) {
+            val messagingStorage = applicationContext.components.analytics.messagingStorage
+            val messages = messagingStorage.getMessages()
+            val nextMessage =
+                messagingStorage.getNextMessage(MessageSurfaceId.NOTIFICATION, messages)
+
+            nextMessage?.apply {
+                // Display the notification if the Message has not been displayed.
+                if (metadata.displayCount == 0) {
+                    messagingStorage.updateMetadata(
+                        Message.Metadata(
+                            id = id,
+                            displayCount = 1,
+                            lastTimeShown = System.currentTimeMillis(),
+                        ),
+                    )
+
+                    NotificationManagerCompat.from(applicationContext).notify(
+                        MESSAGE_TAG,
+                        MESSAGE_NOTIFICATION_ID,
+                        buildNotification(this),
+                    )
+                }
+            }
+        }
+
+        return Result.success()
+    }
+
+    /**
+     * Build the [Notification] with the given [Message] configuration.
+     */
+    private fun buildNotification(message: Message): Notification {
+        with(applicationContext) {
+            val intent = message.getIntent()
+
+            val pendingIntent = PendingIntent.getActivity(
+                this,
+                SharedIdsHelper.getNextIdForTag(this, NOTIFICATION_PENDING_INTENT_TAG),
+                intent,
+                IntentUtils.defaultIntentPendingFlags,
+            )
+
+            return createBaseNotification(
+                this,
+                MARKETING_CHANNEL_ID,
+                message.data.title,
+                message.data.text,
+                pendingIntent,
+            )
+        }
+    }
+
+    companion object {
+        private const val NOTIFICATION_PENDING_INTENT_TAG = "org.mozilla.fenix.message"
+        private const val MESSAGE_TAG = "org.mozilla.fenix.message.tag"
+        private const val MESSAGE_WORK_NAME = "org.mozilla.fenix.message.work"
+
+        /**
+         * Initialize the [Worker] to begin polling Nimbus.
+         */
+        fun setMessageNotificationWorker(context: Context) {
+            val featureConfig = FxNimbus.features.messaging.value()
+            val notificationConfig = featureConfig.notificationConfig
+            val pollingInterval = notificationConfig.pollingInterval.toLong()
```

Comment:
I think we need to change the `pollingInterval` to `15` for the `developer` channel in `messaging.fml.yaml`.

Reply:
It should be doing this already https://github.com/mozilla-mobile/fenix/blob/e6e2b8c8958de093b7ce6e4db029cef1ec622a5e/messaging.fml.yaml? May have been updated while you were reviewing